apps/client/src/components/Questions.js

```diff
@@ -91,20 +103,29 @@ const Questions = ({
         }
       });
     }
-  }, [checker, isActive, questionIndex, onQuestionNext, skipAnsweredQuestions]);
 
-  const onGoToQuestion = useCallback(
-    (questionId) => {
-      // Checker rewinding also needs to work when you already have a conlusion
-      // Go to the specific question in the stack
-      setActiveState("questions");
-      setFinishedState(["conclusion", "questions"], false);
-      setFinishedState("locationResult", true);
+    // @TODO: Refactor this code and move to checker.js
+    // Bug fix in case of refresh: hide already future answered questions (caused by setQuestionAnswers() in withChecker)
+    if (!contactConclusion) {
+      checker.stack.forEach((q, i) => {
+        if (checker.needContactExit(q)) {
+          // Set questionIndex to question that causes the needContactExit()
+          sessionContext.setSessionData([
+            slug,
+            {
+              questionIndex: i,
+            },
+          ]);
 
-      goToQuestion(questionId);
-    },
-    [goToQuestion, setActiveState, setFinishedState]
-  );
+          // Set Contact Conclusion
+          setContactConclusion(true);
+
+          // End loop
+          return;
```

Comment:
Do you mean `break`? Return doesn't end the loop.

Reply:
`return` indeed didn't break the loop. It looked like it when I logged the result, but that turned out to be expected behaviour. 

It turns out [There's no built-in ability to break in forEach](https://stackoverflow.com/a/2641374)

Reply:
So return does nothing here? Can we get rid of it, and the comment too?